cmd/prysmctl/p2p/client.go

```diff
@@ -172,16 +174,52 @@ func (c *client) retrievePeerAddressesViaRPC(ctx context.Context, beaconEndpoint
 	return peers, nil
 }
 
-func (c *client) initializeMockChainService(ctx context.Context) (*mockChain, error) {
+func (c *client) initializeMockChainService(ctx context.Context, forkVersion int) (*mockChain, error) {
 	genesisResp, err := c.nodeClient.GetGenesis(ctx, &emptypb.Empty{})
 	if err != nil {
 		return nil, err
 	}
+
 	currEpoch := slots.ToEpoch(slots.SinceGenesis(genesisResp.GenesisTime.AsTime()))
-	currFork, err := forks.Fork(currEpoch)
-	if err != nil {
-		return nil, err
+	var currFork *pb.Fork
+
+	if forkVersion == -1 {
+		currFork, err = forks.Fork(currEpoch)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		var forkVersionBytes [fieldparams.VersionLength]byte
+
+		// Map runtime/version constants to beacon config version bytes
+		switch forkVersion {
+		case version.Phase0:
+			forkVersionBytes = bytesutil.ToBytes4(params.BeaconConfig().GenesisForkVersion)
+		case version.Altair:
+			forkVersionBytes = bytesutil.ToBytes4(params.BeaconConfig().AltairForkVersion)
+		case version.Bellatrix:
+			forkVersionBytes = bytesutil.ToBytes4(params.BeaconConfig().BellatrixForkVersion)
+		case version.Capella:
+			forkVersionBytes = bytesutil.ToBytes4(params.BeaconConfig().CapellaForkVersion)
+		case version.Deneb:
+			forkVersionBytes = bytesutil.ToBytes4(params.BeaconConfig().DenebForkVersion)
+		case version.Electra:
+			forkVersionBytes = bytesutil.ToBytes4(params.BeaconConfig().ElectraForkVersion)
+		case version.Fulu:
+			forkVersionBytes = bytesutil.ToBytes4(params.BeaconConfig().FuluForkVersion)
+		default:
+			return nil, errors.Errorf("unsupported fork version: %d", forkVersion)
+		}
+
+		// Create fork data from version bytes
+		orderedSchedule := forks.NewOrderedSchedule(params.BeaconConfig())
+		currFork, err = orderedSchedule.ForkFromVersion(forkVersionBytes)
+		if err != nil {
+			return nil, err
+		}
+		log.Infof("Using explicit fork version: %s", version.String(forkVersion))
```

Comment:
```suggestion
		log.Infof("Using explicit fork version %s", version.String(forkVersion))
```
Golang errors shouldn't contain colons because they are used to separate error messages